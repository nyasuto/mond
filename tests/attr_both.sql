-- Price and FX change (USD asset), q constant
INSERT INTO assets (ticker, ccy) VALUES ('VTI','USD');
INSERT INTO fx_rates (date, pair, rate) VALUES ('2025-09-14','USDJPY',140.0);
INSERT INTO fx_rates (date, pair, rate) VALUES ('2025-09-15','USDJPY',145.0);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2025-09-14','VTI',100,200);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2025-09-15','VTI',100,210);

SELECT date,
       ticker,
       round(delta_total, 3) AS total,
       round(delta_price, 3) AS price,
       round(delta_fx, 3)    AS fx,
       round(delta_cross, 3) AS cross,
       round(flow, 3)        AS flow
FROM v_attribution
ORDER BY ticker;

