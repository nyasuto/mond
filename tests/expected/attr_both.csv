date,ticker,total,price,fx,cross,flow
2025-09-15,PORTFOLIO,245000.0,140000.0,100000.0,5000.0,0.0
2025-09-15,VTI,245000.0,140000.0,100000.0,5000.0,0.0

